prow/crier/reporters/github/reporter.go

```diff
@@ -174,11 +177,71 @@ func (c *Client) Report(ctx context.Context, log *logrus.Entry, pj *v1.ProwJob)
 			log.WithError(err).Debug("Could not find PR commit, skipping retries")
 			err = nil
 		}
+		// Always return when there is any error reporting status context.
+		return []*v1.ProwJob{pj}, nil, err
 	}
 
+	// Check if this org or repo has opted out of failure report comments.
+	// This check has to be here and not in ShouldReport as we always need to report
+	// the status context, just potentially not creating a comment.
+	refs := pj.Spec.Refs
+	fullRepo := fmt.Sprintf("%s/%s", refs.Org, refs.Repo)
+	for _, ident := range c.config().GitHubReporter.NoCommentRepos {
+		if refs.Org == ident || fullRepo == ident {
+			return []*v1.ProwJob{pj}, nil, nil
+		}
+	}
+	// Check if this org or repo has opted out of failure report comments
+	toReport := []v1.ProwJob{*pj}
+	for _, ident := range c.config().GitHubReporter.SummaryCommentRepos {
+		if pj.Spec.Refs.Org == ident || fullRepo == ident {
+			toReport, err = pjsToReport(ctx, log, c.lister, pj)
+			if err != nil {
+				return []*v1.ProwJob{pj}, nil, err
+			}
+		}
+	}
+	err = report.ReportComment(ctx, c.gc, c.config().Plank.ReportTemplateForRepo(pj.Spec.Refs), toReport, c.config().GitHubReporter)
+
 	return []*v1.ProwJob{pj}, nil, err
 }
 
+func pjsToReport(ctx context.Context, log *logrus.Entry, lister ctrlruntimeclient.Reader, pj *v1.ProwJob) ([]v1.ProwJob, error) {
+	if len(pj.Spec.Refs.Pulls) != 1 {
+		return nil, nil
+	}
+	// find all prowjobs from this PR
+	selector := map[string]string{}
+	for _, l := range []string{kube.OrgLabel, kube.RepoLabel, kube.PullLabel} {
+		selector[l] = pj.ObjectMeta.Labels[l]
+	}
+	var pjs v1.ProwJobList
+	if err := lister.List(ctx, &pjs, ctrlruntimeclient.MatchingLabels(selector)); err != nil {
+		return nil, fmt.Errorf("Cannot list prowjob with selector %v", selector)
+	}
+
+	latestBatch := make(map[string]v1.ProwJob)
+	for _, pjob := range pjs.Items {
+		if !pjob.Complete() { // Any job still running should prevent from comments
+			return nil, nil
+		}
```

Comment:
We need to be ignoring non-reporting jobs. Maybe filter them out immediately after listing? Please also add a test case for this.